### PR TITLE
Used fixed history to navigate back at the new domain flow

### PR DESCRIPTION
**Pull request by paulopmt1**
## Proposed Changes

* We're using fixed navigation history instead of relying at browser default back history.

Bug:

https://user-images.githubusercontent.com/1044309/217806028-0802a6af-4d3d-47dd-9590-0f844f8177de.mp4

## Testing Instructions

* Follow the video flow and make sure the problem doesn't happen anymore

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/domains/domain-search/index.jsx [hunk 1]
```diff
@@ -277,7 +277,10 @@ class DomainSearch extends Component {
 						<div className="domains__header">
 							{ domainSidebarExperimentUser && (
 								<>
-									<DomainAndPlanPackageNavigation step={ 1 } />
+									<DomainAndPlanPackageNavigation
+										goBackLink={ domainManagementList( selectedSiteSlug ) }
```
Comment (paulopmt1) [inline client/my-sites/domains/domain-search/index.jsx:281]: I didn't find a reliable way to get the previous history from the user, so used the same one we already use on the domains page.

File: client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 export default function DomainAndPlanPackageNavigation( props ) {
 	const translate = useTranslate();
 
 	const goBack = () => {
-		window.history.go( -1 );
+		if ( props.goBackLink ) {
+			page( props.goBackLink );
+		} else {
+			window.history.go( -1 );
+		}
```
Comment (candy02058912) [inline client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx:15]: Another approach is to write the links that we want to go back here, for example something like this.
```
switch (step) {
  case 1:
     // redirect somewhere
   case 2:
    // redirect somewhere
   default:
       window.history.go(-1);
}
```

but I think that's more of a personal preference.

**Review by candy02058912 — APPROVED** [review]
Overall, LGTM, thanks for working on this!
Inevitably we have to give the routes ourselves.

File: client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx (hunk 2 above)
Comment (paulopmt1) [reply client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx:15]: Yeah, I prefer switch case as well when having more than 2 options, it's more elegant! In this case, we should always have two.

Thanks for your review @candy02058912!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7801903

Hi @paulopmt1, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `your domain name`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7801903).

**Pull request merged**